src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject("id", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert manufacturer to DB", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getManufacturerByIdRequest = "SELECT * FROM manufacturers "
+                + "WHERE is_deleted = false AND id = ?;";
+        Manufacturer manufacturer = null;
+        try (Connection connection = connectionDB.getConnect();
+                 PreparedStatement getManufacturerByID = connection
+                         .prepareStatement(getManufacturerByIdRequest)) {
+            getManufacturerByID.setLong(1, id);
+            ResultSet resultSet = getManufacturerByID.executeQuery();
+            while (resultSet.next()) {
+                manufacturer = new Manufacturer();
+                parseResultSet(resultSet, manufacturer);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return Optional.ofNullable(manufacturer);
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String getAllManufacturersRequest = "SELECT * FROM manufacturers WHERE is_deleted = false;";
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        try (Connection connection = connectionDB.getConnect();
+                 PreparedStatement getAllManufacturers = connection
+                         .prepareStatement(getAllManufacturersRequest)) {
+            ResultSet resultSet = getAllManufacturers.executeQuery();
+            while (resultSet.next()) {
+                Manufacturer manufacturer = new Manufacturer();
+                allManufacturers.add(parseResultSet(resultSet, manufacturer));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufactureRequest = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE is_deleted = false AND id  = ?;";
+        try (Connection connection = connectionDB.getConnect();
+                 PreparedStatement updateManufactureStatement =
+                         connection.prepareStatement(updateManufactureRequest)) {
+            updateManufactureStatement.setString(1, manufacturer.getName());
+            updateManufactureStatement.setString(2, manufacturer.getCountry());
+            updateManufactureStatement.setLong(3, manufacturer.getId());
+            Manufacturer oldManufacturer = get(manufacturer.getId()).orElseThrow(SQLException::new);
+            updateManufactureStatement.executeUpdate();
+            return oldManufacturer;
```

Comment:
```suggestion
            return manufacturer;
```
Ты по сути возвращаешь тот же объект что тебе приходит, но при этом ты его заново создаешь и инициализируешь.

Reply:
Не совсем так. Я вначале считываю из БД старое значение по ID на его место пишу новое, а старое возвращаю из метода.

Reply:
ок